test/controllers/users/two_step_verification_resets_controller_test.rb

```diff
@@ -0,0 +1,159 @@
+require "test_helper"
+
+class Users::TwoStepVerificationResetsControllerTest < ActionController::TestCase
+  context "GET edit" do
+    context "signed in as Admin user" do
+      setup do
+        @admin = create(:admin_user)
+        sign_in(@admin)
+      end
+
+      should "display form with submit button & cancel link" do
+        user = create(:two_step_enabled_user)
+
+        get :edit, params: { user_id: user }
+
+        assert_template :edit
+        assert_select "form[action='#{user_two_step_verification_reset_path(user)}']" do
+          assert_select "button[type='submit']", text: "Reset 2-step verification"
+          assert_select "a[href='#{edit_user_path(user)}']", text: "Cancel"
+        end
+      end
+
+      should "authorize access if UserPolicy#reset_2sv? returns true" do
+        user = create(:two_step_enabled_user)
+
+        user_policy = stub_everything("user-policy", reset_2sv?: true)
```

Comment:
Yes, @chrisroos and I have done some similar stubbing in the branch we're working on. Seems better than testing all the combinations, like you say.